src/tests.cpp

```diff
@@ -0,0 +1,96 @@
+#include "quadratic_equation.h"
+
+const int MAX_SIZE_FOR_TEST = 5;
+
+struct CoefficientsAndRootForLinear {
+    double first;
+    double second;
+    double root;
+    int return_roots;
+};
+
+struct CoefficientsAndRootsForQuadratic {
+    double first;
+    double second;
+    double third;
+    double root_first;
+    double root_second;
+    int return_roots;
+};
+
+void TestIsZero(double num[MAX_SIZE_FOR_TEST], bool check_variable) {
+    for (int i = 0; i < MAX_SIZE_FOR_TEST; i++) {
+        if (!(check_variable == IsZero(num[i]))) {
+            printf("[INCORRECT] IsZero(%lf) returned %d; expected %d.\n",
+                    num[i], IsZero(num[i]), check_variable);
+        }
+        printf("[OK] IsZero(%lf) returned %d; expected %d.\n",

```

Comment:
Have you even tried to run this? It prints for all incorrect cases both messages [INCORRECT] and [OK]!